lucille-plugins/lucille-pinecone/src/main/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexer.java

```diff
@@ -78,7 +101,8 @@ private void uploadDocuments(List<Document> documents, String embeddingField, St
 
     if (mode.equalsIgnoreCase("upsert")) {
       List<VectorWithUnsignedIndices> upsertVectors = documents.stream()
-          .filter(doc -> doc.hasNonNull(embeddingField)) // buildUpsertVector would throw an error if embeddings is null
+          // buildUpsertVector would throw an error if embeddings is null,
+          // should add dropDocument stage with stage conditions in pipeline before indexing
```

Comment:
Previously, we had a filter here to filter out documents with no embeddings, but now opted to remove filter to standardize the use case of Indexer, and instead extract the filtering logic to a dropDocument stage with stage conditions.